tables/puppet/puppet_facts.go

```diff
@@ -0,0 +1,74 @@
+package puppet
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+
+	"github.com/osquery/osquery-go/plugin/table"
+	"github.com/pkg/errors"
+)
+
+type puppetFacts struct {
+	Name   string
+	Values map[string]interface{}
+}
+
+func PuppetFactsColumns() []table.ColumnDefinition {
+	return []table.ColumnDefinition{
+		table.TextColumn("node"),
+		table.TextColumn("fact"),
+		table.TextColumn("value"),
+	}
+}
+
+func PuppetFactsGenerate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	var results []map[string]string
+
+	facts, err := getPuppetFacts()
+	if err != nil {
+		return nil, err
+	}
+
+	for factName, factValue := range facts.Values {
+		var value string
+
+		// serialize it as json string if it is a map
+		if v, ok := factValue.(map[string]interface{}); ok {
+			if jsonStr, err := json.Marshal(v); err != nil {
+				return nil, errors.Wrap(err, "marshal to json string")
+			} else {
+				value = string(jsonStr)
+			}
+		} else {
+			// else serialize it as string
+			value = fmt.Sprintf("%v", factValue)
+		}
+
+		result := map[string]string{
+			"node":  facts.Name,
+			"fact":  factName,
+			"value": value,
+		}
+		results = append(results, result)
+	}
+
+	return results, nil
+}
+
+func getPuppetFacts() (*puppetFacts, error) {
+	cmd := exec.Command("puppet", "facts", "--render-as", "json")
```

Comment:
Please use the full path to the puppet binary on each platform rather than relying on the `PATH` environment variable.

Reply:
Thank you for the review, I just updated the PR and using the full path to the puppet binary.